plugins/modules/repository_information.py

```diff
@@ -0,0 +1,165 @@
+
+            "is_template":      if it is template (bool: true or false),
+
+            "archived":         archived status of repository (bool: true or false),
+
+            "language":         language that the repo is using (as string),
+
+            "visibility":       for other users ("private" or "public"),
+
+            "url":              url for repo (as string),
+
+            "default_branch":   branch that repo defaults to (as string),
+
+            "hooks_url":        url for hooks (as string),
+
+            "clone_url":        url for cloning (as string)
+        }
+        {
+            ...
+        }
+        {
+            ...
+        }
+    ]
+'''
+
+from github import Github
+from ansible.module_utils.basic import AnsibleModule
+
+def run_module():
+    module_args = dict(
+        token=dict(type='str', default='No Token Provided.'),
+        organization_name=dict(type='str', default='No Organization Name Provided.'),
+        enterprise_url=dict(type='str', default=''),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        fact=''
+    )
+    #token usage retrieved from module's variables from playbook
+
+    if(module.params['enterprise_url'] == ''):
+        g = Github(module.params['token'])
+    else:
+        g = Github(module.params['token'], base_url=module.params['enterprise_url'])
+
+    output = []
+
+    #organization name retrieved from module's variables from playbook
+    org_name = module.params['organization_name']
+    
+    for repo in g.get_organization(org_name).get_repos():
+        if len(module.params["enterprise_url"]) == 0:
+            current_repo_dict = {
+                "name" : repo.name,
+                "full_name": repo.full_name,
+                "owner": repo.owner.login,
+                "description": repo.description,
+                "private": repo.private,
+                "archived": repo.archived,
+                "language": repo.language,
+                "url": repo.url,
+                "default_branch": repo.default_branch,
+                "hooks_url": repo.hooks_url,
+                "clone_url": repo.clone_url,
+                "visibility": repo.raw_data["visibility"],
+                "is_template": repo.raw_data["is_template"]
+                }
+        else:
+            current_repo_dict = {
+                "name" : repo.name,
+                "full_name": repo.full_name,
+                "owner": repo.owner.login,
+                "description": repo.description,
+                "private": repo.private,
+                "archived": repo.archived,
+                "language": repo.language,
+                "url": repo.url,
+                "default_branch": repo.default_branch,
+                "hooks_url": repo.hooks_url,
+                "clone_url": repo.clone_url
+                }
```

Comment:
The only difference between these blocks are a few attributes.  Set the common attributes in one block of code then add the variations after. e.g.

```
current_repo_dict = {...}
if ...  : 
    curent_repo_dict[...] = ...
```